packages/cli/src/databases/entities/test-run.ee.ts

```diff
@@ -35,4 +35,17 @@ export class TestRun extends WithTimestampsAndStringId {
 
 	@Column(jsonColumnType, { nullable: true })
 	metrics: AggregatedTestRunMetrics;
+
+	// Total number of the test cases, matching the filter condition of the test definition (specified annotationTag)
+	@Column('integer', { nullable: true })
+	total: number;
+
+	// Number of test cases that passed (evaluation workflow was executed successfully)
+	@Column('integer', { nullable: true })
+	passed: number;
+
+	// Number of failed test cases
+	// (any unexpected exception happened during the execution or evaluation workflow ended with an error)
+	@Column('integer', { nullable: true })
+	failed: number;
```

Comment:
Instead of making these nullable, could they be non-nullable with default 0?

Reply:
I was not sure about this for several reasons:
- total number of cases is being calculated after test run entity created (the lifecycle of a test run is: `new` -> fetch past executions -> `running`, save number of matching past executions to `total` -> incrementing `passed` or `failed` while looping over test cases -> `finished`). So `null` and 0 have different meaning: `null` means we don't know the number of test cases yet, and 0 means there was no matching past executions when we tried to run a test.
- storing zeros in `passed` and `failed` might be ok, but still not 100% sure it's right to store exact number while we haven't started test run yet

Reply:
Okay that makes sense! Could we then add a check constraint that ensures the required values are set in correct statuses? I.e.

```sql
(status = 'new' AND total IS NULL AND passed IS NULL AND failed IS NULL) OR
(total IS NOT NULL AND passed IS NOT NULL AND failed IS NOT NULL)
```

Reply:
Agree! Added per-column constraints (as a simplest way to do it for all 3 DB engines)